lib/recording/RecordingEntrySession.js

```diff
@@ -369,16 +427,72 @@ class RecordingEntrySession {
 
     }
 
+ /*   verifyStreamMatchPreviousFlavorSet(newEntryObj) {
+        return qio.exists(this.recordingSessionPath)
+            .then((sessionExist)=> {
+                if (!sessionExist) {
+                    return Q.resolve(true);
+                }
+                else {
+                    let newFlavorNames = _.map(newEntryObj.flavorsObjArray, function(newFlavorDir) {
+                        return newFlavorDir.name;
+                    });
+                    let pattern = /^\d+$/;
+                    return fsUtils.isSameFlavorSet(this.recordingSessionPath, newFlavorNames, [newEntryObj.recordedEntryId], pattern);
+                }
+            })
+            .catch((err) => {
+                this.logger.error(`failed to verify that current flavors match previous recording. Error: ${err}`);
+                // since failed to verify match choosing fallback which is assuming no change in flavor set
+                return Q.resolve(true);
+            })
+    }*/
+
+    verifyFlavorSetMatchPlaylistFlavors(newEntryObj) {
+        let playlistFilePath = path.join(this.recordingSessionPath, 'playlist.json');
+        return qio.read(playlistFilePath)
+            .catch((err) =>{
+                if (err instanceof Error && err.code !== 'ENOENT') {
+                    this.logger.error(`Failed to read [${this.recordingSessionPath}]. Error: ${ErrorUtils.error2string(err)}`);
+                    return Q.reject(err);
+                }
+                else
+                    return Q.resolve();
```

Comment:
how will this work if the folder doesn't exists? (new session)
you pass Q.Resolve, and then next then you try to parse it?